extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.js

```diff
@@ -0,0 +1,174 @@
+import {htmlFor} from '../../../src/static-template';
+
+/**
+ * Creates a tab content, will be deleted when the tabs get implemented.
+ * @param {!Window} win
+ * @param {string} storyUrl
+ * @param {string} name
+ * @return {!Element} the layout
+ */
+export function createTabLogsElement(win, storyUrl) {
+  const element = win.document.createElement('amp-story-dev-tools-tab-logs');
+  element.setAttribute('data-story-url', storyUrl);
+  return element;
+}
+
+/**
+ * Returns a "failed" or "passed" icon from the status
+ * @param {!Element} element
+ * @param {boolean} statusPassed
+ * @return {!Element}
+ */
+function buildStatusIcon(element, statusPassed) {
+  const html = htmlFor(element);
+  if (statusPassed) {
+    return html`<svg xmlns="http://www.w3.org/2000/svg" width="20" height="20">
+      <path
+        d="M10 0C4.48 0 0 4.48 0 10C0 15.52 4.48 20 10 20C15.52 20 20 15.52 20 10C20 4.48 15.52 0 10 0ZM8 15L4 11L5.4 9.6L8 12.2L14.6 5.6L16 7L8 15Z"
+        fill="#2DE561"
+      />
+    </svg>`;
+  } else {
+    return html`<svg xmlns="http://www.w3.org/2000/svg" width="20" height="20">
+      <path
+        xmlns="http://www.w3.org/2000/svg"
+        d="M9.99 0C4.47 0 0 4.48 0 10C0 15.52 4.47 20 9.99 20C15.52 20 20 15.52 20 10C20 4.48 15.52 0 9.99 0ZM11 10C11 10.55 10.55 11 10 11C9.45 11 9 10.55 9 10V6C9 5.45 9.45 5 10 5C10.55 5 11 5.45 11 6V10ZM11 14C11 14.55 10.55 15 10 15C9.45 15 9 14.55 9 14C9 13.45 9.45 13 10 13C10.55 13 11 13.45 11 14Z"
+        fill="#FF5252"
+      />
+    </svg>`;
+  }
+}
+
+const buildLogMessageTemplate = (element) => {
+  const html = htmlFor(element);
+  return html`<div class="i-amphtml-story-dev-tools-log-message">
+    <div>
+      <span class="i-amphtml-story-dev-tools-log-type"></span>
+      <span> at </span>
+      <span class="i-amphtml-story-dev-tools-log-position"></span>
+    </div>
+    <span class="i-amphtml-story-dev-tools-log-description"></span>
+    <a class="i-amphtml-story-dev-tools-log-spec" target="_blank">Learn more</a>
+    <pre class="i-amphtml-story-dev-tools-log-code"></pre>
+  </div>`;
+};
+
+export class AmpStoryDevToolsTabLogs extends AMP.BaseElement {
+  /** @param {!Element} element */
+  constructor(element) {
+    super(element);
+
+    /** @private  {string} */
+    this.storyUrl_ = '';
+  }
+
+  /** @override */
+  buildCallback() {
+    this.storyUrl_ = this.element.getAttribute('data-story-url');
+    this.element.classList.add('i-amphtml-story-dev-tools-tab');
+
+    getInstance().then((validator) => {
+      this.validateUrl_(validator.sandbox.amp.validator, this.storyUrl_);
+    });
+  }
+
+  /**
+   * Validates a URL input, showing the errors on the screen.
+   *
+   * @private
+   * @param {!Validator} validator
+   * @param {string} url
+   */
+  validateUrl_(validator, url) {
+    Services.xhrFor(this.win)
+      .fetchText(url)
+      .then((response) => {
+        userAssert(response.ok, 'Invalid story url');
+        response.text().then((html) => {
```

Comment:
Please use promise chaining, you can return `response.text()` and have another `then` later on. You save one indentation level